components/MDX/UpcomingCalendar.tsx

```diff
@@ -0,0 +1,72 @@
+import type { FC } from 'react';
+
+import Time from '@/components/Common/Time';
+import Link from '@/components/Link';
+import { CALENDAR_NODEJS_ID } from '@/next.calendar.constants.mjs';
+import { getCalendarEvents } from '@/next.calendar.mjs';
+import type { CalendarEvent, ZonedCalendarTime } from '@/types';
+
+type GrouppedEntries = Record<string, Array<CalendarEvent>>;
+
+const isZoned = (d: object): d is ZonedCalendarTime =>
+  'dateTime' in d && 'timeZone' in d;
+
+const getZoomLink = (event: Pick<CalendarEvent, 'description' | 'location'>) =>
+  event.description?.match(/https:\/\/zoom.us\/j\/\d+/)?.[0] ||
+  event.location?.match(/https:\/\/zoom.us\/j\/\d+/)?.[0];
```

Comment:
You should put that in `utils` folder and unit test it. WDYT ?

Reply:
Why? It's used only in one place.

Reply:
Personally, I always separate. But yes, you can leave it there, but I think the unit test is still good practice in all cases.

Reply:
I don't see the value of having a unit test on such a simple regex... It's quite obvious what this regex is matching IMO, or at least what it captures...